### ISSUE TITLE
REFLO install errors regarding PySAM [UPDATED]

### ISSUE THREAD

**Issue by zacharybinger** [watertap-org/watertap-reflo#111]
Recently I tried creating a fresh install of REFLO while attempting to debug other errors i've been running into. After following the install directions from the documentation, many of the tests fail. It seems like the error originates during the import of PySAM

To recreate:
```
git clone https://github.com/watertap-org/watertap-reflo && cd watertap-reflo
conda create --yes --name watertap-reflo-dev python=3.10 && conda activate watertap-reflo-dev
pip install -r requirements-dev.txt

conda activate watertap-reflo-dev
pytest --pyargs watertap_contrib.reflo
```

Current error traceback:
`ImportError while importing test module '/Users/zbinger/watertap-reflo/src/watertap_contrib/reflo/analysis/multiperiod/vagmd_batch/test/test_VAGMD_batch_flowsheet_multiperiod.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
src/watertap_contrib/reflo/analysis/multiperiod/vagmd_batch/test/test_VAGMD_batch_flowsheet_multiperiod.py:23: in <module>
    from watertap_contrib.reflo.analysis.multiperiod.vagmd_batch.VAGMD_batch_flowsheet_multiperiod import (
src/watertap_contrib/reflo/analysis/multiperiod/vagmd_batch/VAGMD_batch_flowsheet_multiperiod.py:43: in <module>
    from watertap_contrib.reflo.costing import REFLOCosting
src/watertap_contrib/reflo/costing/__init__.py:13: in <module>
    from .watertap_reflo_costing_package import (
src/watertap_contrib/reflo/costing/watertap_reflo_costing_package.py:24: in <module>
    from watertap_contrib.reflo.core import PySAMWaterTAP
src/watertap_contrib/reflo/core/__init__.py:14: in <module>
    from .pysam_watertap import PySAMWaterTAP
src/watertap_contrib/reflo/core/pysam_watertap.py:16: in <module>
    import PySAM.Pvsamv1 as pv
E   ImportError: dlopen(/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/Pvsamv1.cpython-310-darwin.so, 0x0002): Library not loaded: @rpath/libortools.9.dylib
E     Referenced from: <94313FC7-2600-3F1F-A5B6-E2A4F2CCAF53> /opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/libSAM_api.so
E     Reason: tried: '/Users/dguittet/Projects/SAM/cmake-build-release/ssc/ssc/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/Projects/SAM/cmake-build-release/ssc/ssc/libortools.9.dylib' (no such file), '/Users/dguittet/Projects/SAM/or-tools/install/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/Projects/SAM/or-tools/install/lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/../../../libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/../../../libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/../../../libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/bin/../lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/bin/../lib/libortools.9.dylib' (no such file), '/usr/local/lib/libortools.9.dylib' (no such file), '/usr/lib/libortools.9.dylib' (no such file, not in dyld cache)Library not loaded: @rpath/libortools.9.dylib
E     Referenced from: <DB7D6C5C-7A80-3401-B65D-7AD9005ACD51> /opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/libssc.so
E     Reason: tried: '/Users/dguittet/Projects/SAM/or-tools/install/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/Projects/SAM/or-tools/install/lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/../../../libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/../../../libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/dguittet/miniconda3/envs/pysam_build_3.10/lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/lib/python3.10/site-packages/PySAM/../../../libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/bin/../lib/libortools.9.dylib' (no such file), '/opt/anaconda3/envs/watertap-reflo-dev/bin/../lib/libortools.9.dylib' (no such file), '/usr/local/lib/libortools.9.dylib' (no such file), '/usr/lib/libortools.9.dylib' (no such file, not in dyld cache)
`

**Comment by lbianchi-lbl** [comment]
- Note that this is seems to be on macOS
- It also looks like @dguittet's username appears in the error message
- Since @dguittet is listed as one of the maintainers https://pypi.org/project/NREL-PySAM/, I'm wondering whether this might be a case of absolute paths "leaking" on the build machine used to generate the macOS wheels https://pypi.org/project/NREL-PySAM/#files

**Comment by zacharybinger** [comment]
NOTE: The errors don't happen when the version of nrel-pysam is reverted back to 3.0.2 like before #107

**Comment by kurbansitterley** [comment]
Closed with #115
